docs/nnx/bridge_guide.md

```diff
@@ -0,0 +1,544 @@
+    self.b = nnx.Param(jax.random.uniform(rngs.params(), (1, out_dim,)))
+
+  def __call__(self, x):
+    return self.dot(x) + self.b
+
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.lazy_init(NNXOuter(64, rngs=nnx.Rngs(0)), x)  # Can fit them into one line too
+w, b = model.dot.params['w'], model.b
+print(w.shape, b.shape)
+y = model(x)
+print(y.shape)
+```
+
+    (32, 64) (1, 64)
+    (4, 64)
+
+
+The Linen weight is already converted to a typical NNX variable, which is a thin wrapper of the actual JAX array value within. Here, `w` is an `nnx.Param` because it belongs to the `params` collection of `LinenDot` module.
+
+We will talk more about different collections and types in the [NNX Variable <-> Linen Collections](#variable-types-vs-collections) section. Right now, just know that they are converted to NNX variables like native ones.
+
+
+```python
+assert isinstance(model.dot.params['w'], nnx.Param)
+assert isinstance(model.dot.params['w'].value, jax.Array)
+```
+
+### NNX -> Linen
+
+To convert an NNX module to Linen, you should forward your creation arguments to `bridge.ToLinen` and let it handle the actual creation process. 
+
+We don't let you create an NNX module of your own because new variables are created eagerly as soon as a NNX module is created. Instead, `bridge.ToLinen` will handle it for you.
+
+An NNX module instance initializes all its variables eagerly when it is created, which consumes memory and compute. On the other hand, Linen modules are stateless, and the typical `init` and `apply` process involves multiple creation of them. To bypass this issue, you should send your creation arguments (instead of a created NNX module) to `bridge.to_linen`, and let it handle the rest.
+
+
+```python
+class NNXDot(nnx.Module):
+  def __init__(self, in_dim: int, out_dim: int, rngs: nnx.Rngs):
+    self.w = nnx.Param(nnx.initializers.lecun_normal()(rngs.params(), (in_dim, out_dim)))
+  def __call__(self, x: jax.Array):
+    return x @ self.w
+
+x = jax.random.normal(jax.random.key(42), (4, 32))
+model = bridge.ToLinen(NNXDot, args=(32, 64))  # <- Pass in the arguments, not an actual module
+variables = model.init(jax.random.key(0), x)
+y = model.apply(variables, x)
+
+print(list(variables.keys()))
+print(variables['params']['w'].value.shape)  # => (32, 64)
+print(y.shape)                               # => (4, 64)
+
+```
+
+    ['nnx', 'params']
+    (32, 64)
+    (4, 64)
+
+
+Note that `ToLinen` modules need to track an extra variable collection - `nnx` - for the static metadata of the underlying NNX module.
+
+
+```python
+# This new field stores the static data that defines the underlying `NNXDot`
+print(type(variables['nnx']['graphdef']))    # => `nnx.graph.NodeDef`
+```
+
+    <class 'flax.nnx.nnx.graph.NodeDef'>
+
+
+You can use a shortcut `bridge.to_linen` to avoid explicitly grouping args and kwargs separately. You only need to use the underlying `bridge.ToLinen()` in some rare cases.
```

Comment:
Maybe we can also forward the other arguments such that you can always use `to_linen`?